language/move-prover/src/prelude.bpl

```diff
@@ -458,8 +455,8 @@ function {:inline} TxnSenderAddress(txn: Transaction): int {
 // Instructions
 
 procedure {:inline 1} Exists(address: Value, t: TypeValue) returns (dst: Value)
+requires is#Address(address);
```

Comment:
I think type checks should be always assumes. They are guaranteed by the type checker to hold; no need for Boogie to prove it for us, it just puts unneeded load onto it. Applies also to below changes.

Reply:
I disagree.  This is not type checking, it's more of a sanity check.  We do need to constrain the types of things to ensure we don't get spurious counterexamples.  This just checks that we did it correctly.  Note that making this change did uncover a couple of interesting issues.  I think that shows that it's useful.

Reply:
I'm good with this but we don't do it near to systematically. Ideally, we would have a flag to the prover to not do this or to do thorough type checking, also in the generated code. This could actually be a great source of "self testing". Perhaps in addition to this, there are more redundant requirements we like to inject. This could be a good intern project, something like "self-verifying verification".

@DavidLDill 
@shazqadeer